unity-ggjj/Assets/Scripts/Menu/CaseSelectMenu.cs

```diff
@@ -0,0 +1,56 @@
+using UnityEngine;
+using System;
+using UnityEngine.UI;
+
+public class CaseSelectMenu : MonoBehaviour
+{
+    [SerializeField] private MenuItem _buttonPrefab;
+    [SerializeField] private Transform _buttonContainer;
+    [SerializeField] private Image _previewImage;
+    [SerializeField] private ChapterSelectMenu _chapterSelectMenu;
+    [SerializeField] private Case[] _cases;
+
+    /// <summary>
+    /// On Awake:
+    /// Subscribe back button to show and hide preview
+    /// Instantiate buttons equal to number of cases
+    /// Set button text
+    /// Subscribe instantiated buttons to update preview image
+    /// Add onClick logic to each button: either start the game or open the chapter select menu
+    /// </summary>
+    private void Awake()
+    {
+        var backButton = GetComponentInChildren<MenuItem>();
+        backButton.OnItemSelect.AddListener(() => _previewImage.color = Color.black);
+        backButton.OnItemDeselect.AddListener(() => _previewImage.color = Color.white);
+
+        foreach (var @case in _cases)
+        {
+            var menuItem = Instantiate(_buttonPrefab, _buttonContainer);
+            menuItem.Text = @case.Name;
+            menuItem.OnItemSelect.AddListener(() => _previewImage.sprite = @case.PreviewImage);
+            var menuOpener = menuItem.GetComponent<MenuOpener>();
+            menuOpener.MenuToOpen = _chapterSelectMenu.Menu;
+            ((Button)menuItem.Selectable).onClick.AddListener(() =>
+            {
+                if (@case.Chapters.Length == 1)
+                {
+                    // No need to show chapters, put logic for starting game here
+                }
+                else
+                {
+                    menuOpener.OpenMenu();
+                    _chapterSelectMenu.Initialise(@case.Chapters, menuOpener);
+                }
```

Comment:
Nit: Rather than commenting this, feel free to throw a `NotImplementedException` here.  
That way, builds crash and the Unity editor prints an error to the console if this point of the code is hit.  
It also reduces nesting, as the `else` becomes redundant.
```suggestion
                if (@case.Chapters.Length == 1)
                {
                    throw new NotImplementedException("Put logic for starting game here");
                }
                
                menuOpener.OpenMenu();
                _chapterSelectMenu.Initialise(@case.Chapters, menuOpener);
```